erica/elster_xml/grundsteuer/elster_data_representation.py

```diff
@@ -3,15 +3,146 @@
-    Eigentuemer as EigentuemerInput, Anteil, Vertreter, Empfangsbevollmaechtigter
+from erica.request_processing.erica_input.v2.grundsteuer_input import GrundsteuerData
+from erica.request_processing.erica_input.v2.grundsteuer_input_eigentuemer import Vertreter, Anteil, Person, \
+    Empfangsbevollmaechtigter, Eigentuemer as EigentuemerInput
+from erica.request_processing.erica_input.v2.grundsteuer_input_gebaeude import Gebaeude as GebaeudeInput, \
+    WeitereWohnraeumeDetails, GaragenAnzahl
 
 """
     The content of the Grundsteuer Nutzdaten XML as its data prepresentation.
     The classes are prefixed with "E" for "Elster".
 """
 
 
+@dataclass
+class EGaragen:
+    E7403171: int  # Anzahl Garagen
+
+    def __init__(self, garagen_anzahl: GaragenAnzahl):
+        self.E7403171 = garagen_anzahl.anzahl_garagen
+
+
+@dataclass
+class EWohnUnter60:
+    E7403131: int = 0  # Anzahl
+    E7403132: int = 0  # Flaeche
+
+    def __init__(self, flaechen: List[int]):
+        for flaeche in flaechen:
+            if flaeche < 60:
+                self.E7403131 += 1
+                self.E7403132 += flaeche
+
+
+@dataclass
+class EWohn60bis100:
+    E7403141: int = 0
+    E7403142: int = 0
+
+    def __init__(self, flaechen: List[int]):
+        for flaeche in flaechen:
+            if 60 <= flaeche < 100:
+                self.E7403141 += 1
+                self.E7403142 += flaeche
+
+
+@dataclass
+class EWohnAb100:
+    E7403151: int = 0
+    E7403152: int = 0
+
+    def __init__(self, flaechen: List[int]):
+        for flaeche in flaechen:
+            if flaeche >= 100:
+                self.E7403151 += 1
+                self.E7403152 += flaeche
+
+
+@dataclass
+class EWeitereWohn:
+    E7403121: int = 0
+    E7403122: int = 0
+
+    def __init__(self, weitere_wohnraeume_details: WeitereWohnraeumeDetails):
+        self.E7403121 = weitere_wohnraeume_details.anzahl
+        self.E7403122 = weitere_wohnraeume_details.flaeche
+
+
+@dataclass
+class EAngDurchschn:
+    Wohn_Unter60: Optional[EWohnUnter60] = None
+    Wohn_60bis100: Optional[EWohn60bis100] = None
+    Wohn_ab100: Optional[EWohnAb100] = None
+    Weitere_Wohn: Optional[EWeitereWohn] = None
+
+    def __init__(self, input_data: GebaeudeInput):
+        flaechen: List[int] = []
+        if input_data.wohnflaeche:
+            flaechen.append(input_data.wohnflaeche.wohnflaeche)
+        elif input_data.wohnflaechen:
+            flaechen.append(input_data.wohnflaechen.wohnflaeche1)
+            flaechen.append(input_data.wohnflaechen.wohnflaeche2)
+
+        wohn_unter60 = EWohnUnter60(flaechen)
+        if wohn_unter60.E7403131 > 0:
+            self.Wohn_Unter60 = wohn_unter60
+
+        wohn_60bis100 = EWohn60bis100(flaechen)
+        if wohn_60bis100.E7403141 > 0:
+            self.Wohn_60bis100 = wohn_60bis100
+
+        wohn_ab100 = EWohnAb100(flaechen)
+        if wohn_ab100.E7403151 > 0:
+            self.Wohn_ab100 = wohn_ab100
```

Comment:
This is I think a bit the other way around than how I pictured the process. So far, the XML nodes just got the information they truly needed. This way, we also won't have to know the specific E-numbers of another class here (e.g.E7403131). So I'd rather construct three lists (unter60 etc) and then give these lists to the sub-nodes.
(and probably also make them english. :x

Reply:
You make a good point. The three EWohn* instances do still just get the information they need to construct their own instances. The limitation here is that we do not want to update the instances in EAngDurchschn as long as the anzahl/flaeche field in each of the three constructed EWohn* instances is 0 (so that the instances with 0 will remain `None` pointers); otherwise they will get serialized into the XML. So we're forced to live a little bit of leakage here in EAngDurchschn. Does that make sense? Or maybe I don't understand your suggestion. Let's talk about it in person if that's the case.

Reply:
Lat's talk about it.

Reply:
This would be my suggestion:

```

@dataclass
class EWohnUnter60:
    E7403131: int = 0  # Anzahl
    E7403132: int = 0  # Flaeche

    def __init__(self, flaechen: List[int]):
        for flaeche in flaechen:
            self.E7403131 += 1
            self.E7403132 += flaeche


@dataclass
class EWohn60bis100:
    E7403141: int = 0
    E7403142: int = 0

    def __init__(self, flaechen: List[int]):
        for flaeche in flaechen:
            self.E7403141 += 1
            self.E7403142 += flaeche


@dataclass
class EWohnAb100:
    E7403151: int = 0
    E7403152: int = 0

    def __init__(self, flaechen: List[int]):
        for flaeche in flaechen:
            self.E7403151 += 1
            self.E7403152 += flaeche


@dataclass
class EWeitereWohn:
    E7403121: int = 0
    E7403122: int = 0

    def __init__(self, weitere_wohnraeume_details: WeitereWohnraeumeDetails):
        self.E7403121 = weitere_wohnraeume_details.anzahl
        self.E7403122 = weitere_wohnraeume_details.flaeche


@dataclass
class EAngDurchschn:
    Wohn_Unter60: Optional[EWohnUnter60] = None
    Wohn_60bis100: Optional[EWohn60bis100] = None
    Wohn_ab100: Optional[EWohnAb100] = None
    Weitere_Wohn: Optional[EWeitereWohn] = None

    def __init__(self, input_data: GebaeudeInput):
        flaechen: List[int] = []
        if input_data.wohnflaeche:
            flaechen.append(input_data.wohnflaeche.wohnflaeche)
        elif input_data.wohnflaechen:
            flaechen.append(input_data.wohnflaechen.wohnflaeche1)
            flaechen.append(input_data.wohnflaechen.wohnflaeche2)

        flaechen_unter60 = []
        flaechen_60bis100 = []
        flaechen_ab100 = []
        
        for flaeche in flaechen:
            if flaeche < 60:
                flaechen_unter60.append(flaeche)
            elif flaeche < 100:
                flaechen_60bis100.append(flaeche)
            else:
                flaechen_ab100.append(flaeche)
        
        self.Wohn_Unter60 = EWohnUnter60(flaechen_unter60)
        self.Wohn_60bis100 = EWohn60bis100(flaechen_60bis100)
        self.Wohn_ab100 = EWohnAb100(flaechen_ab100)

        if input_data.weitere_wohnraeume.has_weitere_wohnraeume:
            self.Weitere_Wohn = EWeitereWohn(input_data.weitere_wohnraeume_details)

```

Reply:
Done.